chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/search/SearchRequest.java

```diff
@@ -0,0 +1,276 @@
+package gov.healthit.chpl.domain.search;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.util.StringUtils;
+
+public class SearchRequest implements Serializable {
+    private static final long serialVersionUID = 1179207628639701580L;
+
+    public static final String CERTIFICATION_DATE_SEARCH_FORMAT = "yyyy-MM-dd";
+    public static final int MAX_PAGE_SIZE = 100;
+    public static final String ORDER_BY_DEVELOPER = "developer";
+    public static final String ORDER_BY_PRODUCT = "product";
+    public static final String ORDER_BY_VERSION = "version";
+    public static final String ORDER_BY_CERTIFICATION_EDITION = "certificationEdition";
+    public static final String ORDER_BY_CERTIFICATION_BODY = "certificationBody";
+    
+    String searchTerm = null;
+
+    // search for any of these
+    Set<String> certificationStatuses = new HashSet<String>();
+    // search for any of these
+    Set<String> certificationEditions = new HashSet<String>();
+    
+    Set<String> certificationCriteria = new HashSet<String>();
+    //AND or OR the certification criteria together
+    SearchSetOperator certificationCriteriaOperator = SearchSetOperator.OR; 
+    
+    Set<String> cqms = new HashSet<String>();
+    //AND or OR the cqms together
+    SearchSetOperator cqmsOperator = SearchSetOperator.OR; 
+    
+    // search for any of these
+    Set<String> certificationBodies = new HashSet<String>();
+
+    String developer = null;
+    String product = null;
+    String version = null;
+    String practiceType = null;
+    String certificationDateStart = null;
+    String certificationDateEnd = null;
+    SurveillanceSearchFilter surveillance = new SurveillanceSearchFilter();
+    
+    String orderBy = ORDER_BY_PRODUCT;
+    Boolean sortDescending = false;
+    Integer pageNumber = 0;
+    Integer pageSize = 20;
+
```

Comment:
Do the pageNumber/pageSize parameters need to be cleaned / checked for "isNaN" status?

Reply:
Page size is checked for not being greater than 100. Because they are declared as Integers in this object and in the GET request params, spring will automatically send back a bad request error if the user passes in something it can't be converted to integer.